source/server-discovery-and-monitoring/tests/rs/discover_hidden.yml

```diff
@@ -0,0 +1,50 @@
+description: "Hidden discovered"
```

Comment:
Is this a newly added test? What is it testing and why can it not be tested with the replicaSet option as well?

Reply:
On a closer look, this seems to duplicate the discover RSOther test. It should probably be removed?

Reply:
The outcomes are the same but server responses are different, because several server responses are treated by the driver as mapping to RSOther state. I added the test with replicaSet option.